src/store/mod.rs

```diff
@@ -123,11 +125,19 @@ impl StoreConfig {
     pub fn default_cached_above_base_layer(leafs: usize, branches: usize) -> usize {
         let height = get_merkle_tree_height(leafs, branches);
         if height < 2 {
-            0
-        } else if height <= DEFAULT_CACHED_ABOVE_BASE_LAYER {
+            return 0;
+        }
+
+        let default_height = match branches {
+            2 => DEFAULT_CACHED_ABOVE_BASE_LAYER_BINARY,
```

Comment:
For binary, I think that value should be higher.  It looks like if a binary tree has a height of 2, we return cached_above_base as 2 (which will fail since there's nothing to compact) -- we need 0 in that case.

Reply:
^ Nevermind ;-)